src/libData/AccountData/AccountStore.cpp

```diff
@@ -497,8 +499,30 @@ bool AccountStore::UpdateAccountsTemp(const uint64_t& blockNum,
   unique_lock<mutex> g2(m_mutexDelta, defer_lock);
   lock(g, g2);
 
-  return m_accountStoreTemp->UpdateAccounts(blockNum, numShards, isDS,
-                                            transaction, receipt, error_code);
+  bool isEvm{false};
+
+  if (Transaction::GetTransactionType(transaction) ==
+      Transaction::CONTRACT_CREATION) {
+    isEvm = EvmUtils::isEvm(transaction.GetCode());
+  } else if (Transaction::GetTransactionType(transaction) ==
+             Transaction::CONTRACT_CALL) {
+    Account* contractAccount = this->GetAccount(transaction.GetToAddr());
```

Comment:
I think you should use `GetAccountTemp` here. probably there's no difference at the beginning. But it is more reliable. This is what's happening when the txn is processed in UpdateAccounts as it was before the change.